presto-main/src/main/java/com/facebook/presto/operator/DriverContext.java

```diff
@@ -194,6 +195,7 @@ public void transferMemoryToTaskContext(long bytes)
     {
         ListenableFuture<?> future = pipelineContext.reserveMemory(bytes);
         memoryReservation.getAndAdd(bytes);
+        peakMemoryReservation.getAndAccumulate(memoryReservation.get(), Math::max);
```

Comment:
There's a potential race here. You should change the above call to a `addAndGet` and then use the result of that here, instead of calling `get()` again